xmrstak/backend/nvidia/nvcc_code/cuda_extra.cu

```diff
@@ -582,18 +591,19 @@ extern "C" int cuda_get_deviceinfo(nvid_ctx* ctx)
 
 		int* tmp;
 		cudaError_t err;
+		#define MSG_CUDA_FUNC_FAIL "WARNING: skip device - %s failed\n"
```

Comment:
some here please do not use defines here.

Reply:
Weird, ok.  #define is functionally identical to duplicated static strings by the time it compiles, with bonus of never being able to screw up the message separately...  allocating string on stack is wasteful so the define was the best way out while not having dupe message strings.  Also maybe sort of thinking into the future and translation packs which would essentially require defines.